legend-engine-xt-documentStore-executionPlan-connection/src/main/java/org/finos/legend/engine/plan/execution/stores/nonrelational/MongoDbResource.java

```diff
@@ -217,7 +254,7 @@ public Response insertCollectionItemAsJson(@QueryParam("collectionName") String
 
     private MongoDatabase getDefaultDB()
     {
-        return this.mongoDbClient.getMongoDBClient().getDatabase(DEFAULT_DATABASE_NAME);
+        return this.mongoDbClient.mongoClient.getDatabase(DEFAULT_DATABASE_NAME);
```

Comment:
should we make this this.NonRelationalClient.getClient().getDatabase()
wondering with all Non-Relational naming do we want to make the objet MongoDbClient?   Ok either way, but would be good to think about it.

Reply:
I would keep this as MongoDbClient as that is clear and not tied down to anything else. ( can also see us having multiple clients in the future ie. DynamoDbClient ) Happy to have a NonRelationalClientProvider abstraction though

Reply:
NonRelationalClient is now the generic interface, abstract is simply so we don't duplicate logic of InMem vs ExternalMongoClients for nativeQueryExecution, happy to move those to both these implementations if we prefer